File: exceptions.py
"""Excepciones personalizadas del sistema de biblioteca."""

from __future__ import annotations


class LibroNoEncontradoError(Exception):
    """Se lanza cuando el libro no existe."""


class UsuarioSuspendidoError(Exception):
    """Se lanza cuando el usuario está suspendido o no existe."""


class SinStockError(Exception):
    """Se lanza cuando el libro no tiene copias disponibles."""

File: exceptions.php
<?php
declare(strict_types=1);

/**
 * Archivo de excepciones personalizadas del sistema de biblioteca.
 *
 * Estas clases permiten diferenciar errores de negocio
 * de errores técnicos o de validación general.
 */

/**
 * Se lanza cuando un libro no existe en la base de datos.
 */
class LibroNoEncontradoError extends Exception
{
}

/**
 * Se lanza cuando un usuario suspendido intenta realizar un préstamo.
 */
class UsuarioSuspendidoError extends Exception
{
}

/**
 * Se lanza cuando no hay copias disponibles de un libro.
 */
class SinStockError extends Exception
{
}
?>